content/04.results.md

```diff
@@ -3,11 +3,62 @@
+Interestingly, the discriminative model does well in predicting the CbG edge type, when using the generative model trained on GiG label functions.
+This trend doesn't hold for the reverse as the discriminative model slightly underperforms the generative model.
+Lastly for CbG and GiG using all label functions didn't aid in the discriminative model performance.
 
-## Random Noise Generative Model
-place the results of random label function experiment
+![
+Grid of Area Under the Receiver Operating Curve (AUROC) scores for each discriminative model trained using generated labels from the generative models.
+The rows depict the relationship each model is trying to predict and the columns are the relationship specific sources each label function was sampled from. 
+For example, the top-left most square depicts the discriminator model predicting Disease associates Gene (DaG) sentences, while randomly sampling label functions designed to predict the DaG relationship. 
+The square towards the right depicts the discriminative model predicting DaG sentences, while randomly sampling label functions designed to predict the Compound treats Disease (CtD) relationship.
+This pattern continues filling out the rest of the grid.
+The last most column consists of pooling every relationship specific label function and proceeding as above.
+](https://raw.githubusercontent.com/greenelab/snorkeling/master/figures/label_sampling_experiment/disc_performance_test_set_auroc.png){#fig:discriminative_model_auroc}
+
+![
+Grid of Area Under the Receiver Operating Curve (AUROC) scores for each discriminative model trained using generated labels from the generative models.
+The rows depict the relationship each model is trying to predict and the columns are the relationship specific sources each label function was sampled from. 
+For example, the top-left most square depicts the discriminator model predicting Disease associates Gene (DaG) sentences, while randomly sampling label functions designed to predict the DaG relationship. 
+The square towards the right depicts the discriminative model predicting DaG sentences, while randomly sampling label functions designed to predict the Compound treats Disease (CtD) relationship.
+This pattern continues filling out the rest of the grid.
+The last most column consists of pooling every relationship specific label function and proceeding as above.
+](https://raw.githubusercontent.com/greenelab/snorkeling/master/figures/label_sampling_experiment/disc_performance_test_set_auprc.png){#fig:discriminative_model_auprc}
+
+### Discriminative Model Calibration
+Deep learning models can be overconfident in their predictions. 
+Based on this notion we calibrated the best performing discriminative model using a temperature scaling algorithm.
+For Disease associates Gene (DaG) and Compound treats Disease (CtD) the model is calibrated for high prob predictions.
```

Comment:
```suggestion
We examined the calibration of our existing model (Figure {@fig:discriminative_model_calibration}, blue line).
We found that the DaG and CtG edge types were, though not perfectly calibrated, were somewhat aligned with the ideal calibration lines.
The CbG and GiG edges were poorly calibrated and increasing model certainty did not always lead to an increase in precision.
Applying the calibration algorithm (orange line) did not appear to bring predictions in line with the ideal calibration line, but did capture some of the uncertainty in the GiG edge type.
For this reason we use the measured precision instead of the predicted probabilities when determining how many edges could be added to existing knowledge bases with specified levels of confidence.
```